src/java.base/share/classes/java/lang/Enum.java

```diff
@@ -166,13 +168,29 @@ public final boolean equals(Object other) {
         return this==other;
     }
 
+    /**
+     * The hash code of this enumeration constant.
+     *
+     * @implNote Once initialized, the field value does not change.
+     * Hotspot's identity hash code generation also never returns zero
```

Comment:
Trivial: it's camel case in HotSpot

For example: https://en.wikipedia.org/wiki/HotSpot_(virtual_machine)